addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -240,6 +252,7 @@ private AutomationProgress runPlan(LinkedHashMap<?, ?> envData, ArrayList<?> job
                 break;
             }
             progress.info(Constant.messages.getString("automation.info.jobend", job.getType()));
+            job.logFailedTestsToProgress(progress);
```

Comment:
I think this should be before the `env.isTimeToQuit()` check, otherwise the next job would be run before checking whether the previous one had any failed tests logged to the progress which should have lead to quitting the plan.